keras_cv/layers/preprocessing/center_crop.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv import bounding_box
+from keras_cv.layers.preprocessing.base_image_augmentation_layer import (
+    BaseImageAugmentationLayer,
+)
+from tensorflow import keras
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class CenterCrop(BaseImageAugmentationLayer):
+    """A preprocessing layer which crops images.
+
+    The input images should have values in the `[0-255]` or `[0-1]` range.
+
+    Input shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+
+    Output shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+
+    Args:
+        height: Integer, the height of the output shape.
+        width: Integer, the width of the output shape.
+
+    Call arguments:
+        inputs: Tensor representing images of shape
+            `(batch_size, width, height, channels)`, with dtype tf.float32 / tf.uint8,
+            ` or (width, height, channels)`, with dtype tf.float32 / tf.uint8
+        training: A boolean argument that determines whether the call should be run
+            in inference mode or training mode. Default: True.
+
+    Usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    channel_shuffle = keras_cv.layers.CenterCrop(10, 10)
+    augmented_images = channel_shuffle(images)
+    ```
+    """
+
+    def __init__(self, height: int, width: int, bounding_box_format=None, seed=None, **kwargs):
+        super().__init__(seed=seed, **kwargs)
+        self.height = height
+        self.width = width
+        self.bounding_box_format = bounding_box_format
+        self.base_layer_ = keras.layers.CenterCrop(self.height, self.width)
+
+    def augment_image(self, image, transformation=None, **kwargs):
+        image = self.base_layer_(image)
+        return image
+
+    def get_random_transformation(
+            self,
+            image=None,
+            label=None,
+            bounding_boxes=None,
+            keypoints=None,
+            segmentation_mask=None,
+    ):
+        image_shape = tf.shape(image)
+        return image_shape[-3], image_shape[-2]
+
+    def _transform_bounding_boxes(self, bounding_boxes, transformation):
```

Comment:
do you need to filter boxes that has no overlap with the center crop region?

Reply:
This is handled by `bounding_box.clip_to_image` inside `augment_bounding_boxes,` as seen in the unit test it seems to work

Reply:
Yep!  clip_to_image() does in fact filter out of bounds bounding boxes!

Reply:
you might also need to add a call to `bounding_box.filter_sentinels()` to go from dense -1 padded tensor -> ragged